lib/perl/Genome/WorkflowBuilder/Command.pm

```diff
@@ -114,9 +114,16 @@ sub _get_ptero_shortcut_method {
 sub _get_ptero_execute_method {
     require Ptero::Builder::Job;
 
-
     my $self = shift;
     my $log_dir = shift;
+
+    my $guard;
+    if (my $lsb_sub_additional = Genome::Config::get('lsb_sub_additional')) {
+        my $old = $ENV{LSB_SUB_ADDITIONAL};
+        $guard = Scope::Guard->new(sub { $ENV{LSB_SUB_ADDITIONAL} = $old; });
+        $ENV{LSB_SUB_ADDITIONAL} = $lsb_sub_additional;
+    }
+
     my $ptero_lsf_parameters = $self->_get_ptero_lsf_parameters();
```

Comment:
In several other language APIs, I have seen a "default value" concept for retrieving values.  Another option here could be:

```perl
local $ENV{LSB_SUB_ADDITIONAL} = Genome::Config::get('lsb_sub_additional', default => $ENV{LSB_SUB_ADDITIONAL});
```

For extra flexibility for situations where default values have costly derivation logic:

```perl
local $ENV{LSB_SUB_ADDITIONAL} = Genome::Config::get('lsb_sub_additional', default => sub { $ENV{LSB_SUB_ADDITIONAL} });
```